Assets/_Project/Prototype Pattern/Scripts/Player/InputHandler.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections;
+using PrototypePattern.Input;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace PrototypePattern.Player
+{
+    public class InputHandler : MonoBehaviour
+    {
+        private PlayerController _player;
+        private PlayerControls _playerControls;
+
+        private InputAction _attack;
+        private InputAction _move;
+
+        private Rigidbody2D _rigidBody2D;
+
+        [SerializeField] private Vector2 direction;
+        [SerializeField] private float _speed = 20f;
+        [SerializeField] private bool _isInKnockback = false;
+
+        private void Start()
+        {
+            _rigidBody2D = GetComponent<Rigidbody2D>();
+            _player = GetComponent<PlayerController>();
+        }
+        private void FixedUpdate()
+        {
+            if (!_player.Invencible)
+            {
+                _rigidBody2D.velocity = new Vector2(direction.x * _speed, direction.y * _speed);
```

Comment:
Why not multiply the vector directly?

```suggestion
                _rigidBody2D.velocity = _direction * _speed;
```